core/src/test/java/org/apache/iceberg/rest/RESTCatalogAdapter.java

```diff
@@ -105,6 +105,8 @@ enum HTTPMethod {
 
   enum Route {
     TOKENS(HTTPMethod.POST, "v1/oauth/tokens", null, OAuthTokenResponse.class),
+    SEPARATE_AUTH_TOKENS_URI(
```

Comment:
This looks like it was added to facilitate with the tests, but the CatalogAdapter servers a larger purpose than just the tests.  I'm not confident we want to handle it like this.

Reply:
That makes sense @danielcweeks . I'm lacking the context on what that other purpose is, since it's a class that's implemented and used within the test suite just by looking at the code base... 

Would it make sense to check the responseType to see if it is of OAuthTokenResponse.class type in the default case instead? This means that any POST request on the RestCatalogAdapter that has OAuthTokenResponse.class as the responseType will be interpreted as an OAuth2 request to an external uri

Reply:
I think this will work for now to satisfy the test conditions.  It also looks like this wouldn't be a resolvable route given that it's a full URL that wouldn't match the adapter.